SpokestackTray/Controllers/SpeechController.swift

```diff
@@ -0,0 +1,492 @@
+                    
+                    let metaDataURL: URL = URL.spsk_documentsDirectory.appendingPathComponent(url.lastPathComponent)
+                    try? response.data.write(to: metaDataURL)
+                    
+                    return metaDataURL
+                }
+                .eraseToAnyPublisher()
+    }
+
+    lazy private var pipeline: SpeechPipeline = {
+
+         return try! SpeechPipelineBuilder()
+                    .addListener(self)
+                    .setDelegateDispatchQueue(DispatchQueue.main)
+                    .useProfile(.tfLiteWakewordAppleSpeech)
+                    .setProperty("tracing", Trace.Level.PERF)
+                    .setProperty("detectModelPath", URL.spsk_documentsDirectory.appendingPathComponent("detect.tflite").path)
+                    .setProperty("encodeModelPath", URL.spsk_documentsDirectory.appendingPathComponent("encode.tflite").path)
+                    .setProperty("filterModelPath", URL.spsk_documentsDirectory.appendingPathComponent("filter.tflite").path)
+                    .build()
+    }()
+    
+    // MARK: Initializers
+    
+    override init() {
+        
+        super.init()
+        tts = TextToSpeech(self, configuration: speechConfiguration)
+    }
+    
+    /// Starts the `SpeechPipeline`
+    /// - Returns: Void
+    func start() -> Void {
+
+        if self.startedPublisher.value == true || self.listeningPublisher.value == true {
+            return
+        }
+        
+        self.pipeline.start()
+    }
+    
+    /// Stops the `SpeechPipeline`
+    /// - Returns: Void
+    func stop() -> Void {
+        self.pipeline.stop()
+    }
+    
+    func activate() -> Void {
+
+        if self.listeningPublisher.value == false {
+            self.pipeline.activate()
+        }
+    }
+    
+    func deactive() -> Void {
+        
+        if self.listeningPublisher.value == true {
+            self.pipeline.deactivate()
+        }
+    }
+    
+    func synthesizeSpeech(_ input: String) -> Void {
+    
+        if self.isSilent {
+            return
+        }
+
+        self.tts?.speak(TextToSpeechInput(input, voice: configuration.voice, inputFormat: configuration.ttsFormat))
+    }
+
+    func initializeNLU() -> Void {
+        
+        self.speechConfiguration.nluModelPath = URL.spsk_documentsDirectory.appendingPathComponent("nlu.tflite").path
+        self.speechConfiguration.nluVocabularyPath = URL.spsk_documentsDirectory.appendingPathComponent("vocab.txt").path
+        self.speechConfiguration.nluModelMetadataPath = URL.spsk_documentsDirectory.appendingPathComponent("metadata.json").path
+        self.nlu = try! NLUTensorflow(self, configuration: self.speechConfiguration)
+    }
+}
+
+extension SpeechController: NLUDelegate {
+ 
+    func classification(result: NLUResult) {
+        
+        guard let intentResult: IntentResult = configuration.handleIntent?(result.intent, result.slots, result.utterance) else {
+            return
```

Comment:
it would be good to signal to the client that an intent could not be derived from the user utterance.

Reply:
agreed